scripts/set_final_strain_name.py

```diff
@@ -24,15 +33,20 @@ def replace_name_recursive(node, lookup):
     args = parser.parse_args()
 
     metadata = read_metadata(args.metadata, id_columns=args.metadata_id_columns)
-    name_lookup = {}
-    for ri, row in metadata.iterrows():
-        strain_id = row.name
-        name_lookup[strain_id] = args.display_strain_name if pd.isna(row[args.display_strain_name]) else row[args.display_strain_name]
 
-    with open(args.input_auspice_json, 'r') as fh:
-        data = json.load(fh)
+    if args.display_strain_name in metadata.columns:
+        name_lookup = {}
+        for ri, row in metadata.iterrows():
+            strain_id = row.name
+            name_lookup[strain_id] = args.display_strain_name if pd.isna(row[args.display_strain_name]) else row[args.display_strain_name]
+
+        with open(args.input_auspice_json, 'r') as fh:
+            data = json.load(fh)
 
-    replace_name_recursive(data['tree'], name_lookup)
+        replace_name_recursive(data['tree'], name_lookup, args.metadata_id_columns[0])
```

Comment:
The hard-coded index really stood out to me here, but maybe can be removed if the work around is no longer needed.